### PR TITLE
Add a *CHANGELOG.md* file

### PR DESCRIPTION
**Pull request by krader1961**
Most projects have something analoguous to the *CHANGELOG.md* file
in this commit. The intent is to record significant changes from one
stable release to the next. This initial commit is going to be incomplete
because it was created long after the AT&T AST project was open sourced.
But hopefully it captures the significant changes made when maintenance
of the project resumed.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
+# ksh x.y.z (version TBD, this is a work in progress)
+
+Starting in May 2017 maintenance of the Korn shell (`ksh`) resumed with
```
Comment (siteshwar) [inline CHANGELOG.md]: I received commit access at end of June. I started merging changes (that dates back before I got commit access) from my fork in July.

File: CHANGELOG.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+the merging of some fixes from RedHat by Siteshwar Vashisht. In October
```
Comment (siteshwar) [inline CHANGELOG.md]: 'RedHat' -> 'Red Hat'

File: CHANGELOG.md [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+minor or major release is TBD.
+
+## Deprecations
+
+None at this time.
+
+## Notable non-backward compatible changes
+
+- Support for the UWIN environment has been removed (issue #284).
+- The `SHOPT_FIXEDARRAY` feature has been removed. This should not break any
```
Comment (siteshwar) [inline CHANGELOG.md]: `SHOPT_FIXEDARRAY` was enabled in Fedora.

Comment (krader1961) [reply CHANGELOG.md]: How likely is it anyone is using that feature? See issue #234 for why I am skeptical that anyone is using it and why it shouldn't be used.

**Comment by krader1961** [comment]
Merged with feedback.

**Pull request closed without merging**